backend/bitboxbase/rpcclient/rpcclient.go

```diff
@@ -0,0 +1,174 @@
+	LightningAlias string  `json:"lightningAlias"`
+}
+
+// RPCClient implements observable blockchainInfo.
+type RPCClient struct {
+	observable.Implementation
+	sampleInfo          *SampleInfoResponse
+	log                 *logrus.Entry
+	address             string
+	bitboxBaseConfigDir string
+
+	bitboxBaseNoiseStaticPubkey   []byte
+	channelHash                   string
+	channelHashAppVerified        bool
+	channelHashBitBoxBaseVerified bool
+	sendCipher, receiveCipher     *noise.CipherState
+
+	weHaveQuitChan chan<- struct{}
+
+	//rpc stuff
+	client       *rpc.Client
+	bitboxBaseID string
+}
+
+// NewRPCClient returns a new bitboxbase rpcClient.
+func NewRPCClient(address string, bitboxBaseConfigDir string) *RPCClient {
+	rpcClient := &RPCClient{
+		log:                 logging.Get().WithGroup("bitboxbase"),
+		address:             address,
+		sampleInfo:          &SampleInfoResponse{},
+		bitboxBaseConfigDir: bitboxBaseConfigDir,
+	}
+	return rpcClient
+}
+
+// Connect starts the websocket go routine, first checking if the middleware is reachable,
+// then establishing a websocket connection, then authenticating and encrypting all further traffic with noise.
+func (rpcClient *RPCClient) Connect(address string, bitboxBaseID string) error { //, bitboxBaseEvent chan proto.Message, apiMap map[string]chan *basemessages.BitBoxBaseOut) error {
+	rpcClient.bitboxBaseID = bitboxBaseID
+	response, err := http.Get("http://" + address + "/")
+	if err != nil {
+		rpcClient.log.Println("No response from middleware", err)
+		return err
+	}
+
+	if response.StatusCode != http.StatusOK {
+		rpcClient.log.Println("Received http status code from middleware other than 200")
+		return err
+	}
+
+	bodyBytes, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		rpcClient.log.Println("Body Bytes not read properly")
+		return err
+	}
+	_, err = regexp.MatchString("OK!", string(bodyBytes))
+	if err != nil {
+		return errp.New("rpcClient: Unexpected Response Body Bytes")
+	}
+	if err = response.Body.Close(); err != nil {
+		return errp.New("rpcClient: Failed to close Get Env response")
+	}
+	rpcClient.log.Printf("connecting to base websocket")
+	ws, _, err := websocket.DefaultDialer.Dial("ws://"+rpcClient.address+"/ws", nil)
+	if err != nil {
+		return errp.New("rpcClient: failed to create new websocket client")
+	}
+	if err = rpcClient.initializeNoise(ws, bitboxBaseID); err != nil {
+		return err
+	}
+
+	sendChan, weHaveQuitChan, receiveChan, _ := rpcClient.runWebsocket(ws)
+	rpcClient.weHaveQuitChan = weHaveQuitChan
+	rpcClient.client = rpc.NewClient(&rpcConn{receiveChan, sendChan, weHaveQuitChan})
+	return nil
+}
+
+//Stop shuts down the websocket connection with the base
+func (rpcClient *RPCClient) Stop() {
+	close(rpcClient.weHaveQuitChan)
+}
+
+// GetEnv makes a synchronous rpc call to the base and return the network type and electrs rpc port
+func (rpcClient *RPCClient) GetEnv() (network, electrsRPCPort string) {
```

Comment:
don't name the return values if you don't use the vars in the body. imho very rarely needed.